### PR TITLE
Added active message readability

### LINKED ISSUE #126: Active Message Reading
**Who are you?**
CSE Dev Team

**What feature do you want?**
Make the bot read all messages and follow a/multiple channel(s)

Stretch goal: Reply "That looks interesting" to every message that contains a '?'
Next Stretch goal: Make a command that can turn the listening on or off for a specific channel

Super Stretch goal: Figure out ratelimits and whether or not this will break the bot
What would happen if it was listening everywhere, what would happen to the ratelimit

**Why do you want this feature?**
This would help the FAQ command to be an active command that the bot automatically deals with rather than us manually doing it

**Additional context**
Any questions about this talk to Matt Kijowski, he was the one that started this conversation and this idea

### PR DESCRIPTION
**Pull request by ReginPotter**
Added active message reading in selected channels and added the ability to turn this function on/off

# Description

Allowed the bot to actively read messages in selected channels.
Also added the ability to turn this function on or off

## Issues

Closes #126 

## Type of change

Select one or more of the following:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (describe below)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. List any edge cases you tested.
If you come up with any edge cases you didn't test while writing this PR, cancel the PR and test again.

- [x] Test A
       - Type any message without a ? included
       Nothing should happen
 
- [x] Test B
       - Type any message with a ? included
       Nothing should happen

- [x] Test C
       - '-faq'
       - Type any message without a ? included
       Nothing should happen
 
- [x] Test D
       - '-faq'
       - Type any message with a ? included
       The bot should respond with "That looks interesting"

- [x] Test E
       - '-faq'
       - Type any message with/without a ? included 
       - '-faq'
       - Type any message with/without a ? included
       Nothing should happen

# Checklist:

- [X] All local commits have been pushed to remote
- [X] All changes on the base branch have been merged into this branch, either by rebase or merge
- [X] My code is [PEP-8](https://pep8.org/) compliant (excluding maximum line length, keep that to 100ish characters)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added [Google Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for all new functions/methods
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: Cogs/ServerManagement.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from http.client import HTTPException
```
Comment (mbozada) [inline Cogs/ServerManagement.py:1]: Is this line still needed?

File: Cogs/ServerManagement.py [hunk 2]
```diff
@@ -339,3 +341,48 @@ async def on_button_click(self, res):
                 else:
                     await member.add_roles(role)
                     await res.respond(type=InteractionType.ChannelMessageWithSource, content=f'Gave you the {role.name} role!')
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        """
+        Checks if the author of the message was a bot and if it was sent in a correct channel
```
Comment (mbozada) [inline Cogs/ServerManagement.py:348]: I think this documentation can be made more precise.

Something like:
"Listens to every message on the server, responds if there's a '?' and the channel has FAQ enabled."

File: Cogs/ServerManagement.py [hunk 3]
```diff
@@ -339,3 +341,48 @@ async def on_button_click(self, res):
...
+            # Logging
+            await log(self.bot, f'{ctx.author} has disabled FAQ for the {ctx.channel} channel')
+            return
+        
+        # If the channel is not in the list add it to the end
+        await ctx.reply("FAQ has been enabled for this channel!")
+        self.faq_channels.append(ctx.channel)
+
+        # Logging
+        await log(self.bot, f'{ctx.author} has enabled FAQ for the {ctx.channel} channel')
```
Comment (mbozada) [inline Cogs/ServerManagement.py:388]: Remember to have an empty space at the end of your files.

File: Cogs/StudentCommands.py [hunk 4]
```diff
@@ -228,28 +228,4 @@ async def attendance(self, ctx):
...
-            color=discord.Color.blue())
-        embed.set_author(name="CSE Dev Team", url="http://www.wright.edu/", icon_url="https://d28htnjz2elwuj.cloudfront.net/wp-content/uploads/2019/04/30105443/Wright-State-University-400x400.jpg")
-
-        embed.add_field(name="How do I add a class?", value="Go to the appropriate 'class selection' channel and click the button of your class", inline=False)
-        embed.add_field(name="Need help with coding?", value="If you are in a class higher than 1181, ask questions in your class chat otherwise go to the appropriate help room", inline=False)
-        embed.add_field(name="What can I send in chats?", value="This server is an official discord server of Wright State therefore please do not send any code that pertains to any project/homework in all channels. Please also refrain from any language you ", inline=False)
-        embed.add_field(name="Why won't my role add?", value="Please make sure you have clicked the button. If you have give discord a moment, sometimes discord is slow.", inline=False)
-        await ctx.send(embed=embed)
+            await ctx.message.channel.send(f"You must be in a voice channel to use this command.")
```
Comment (mbozada) [inline Cogs/StudentCommands.py:231]: Whitespace.

**Review by mbozada — CHANGES_REQUESTED** [review]
Good job! It works great, I just had a couple small things.

I'm still curious about if we'll run into rate limit issues, but that can be a separate issue.

File: Cogs/ServerManagement.py (hunk 2 above)
Comment (ReginPotter) [reply Cogs/ServerManagement.py]: This has been changed

File: Cogs/ServerManagement.py (hunk 3 above)
Comment (ReginPotter) [reply Cogs/ServerManagement.py]: Empty space has been added

Comment (mbozada) [reply Cogs/ServerManagement.py]: Still looks weird :man_shrugging:

**Review by mbozada — APPROVED** [review]
Looks nice, works well.
